### ISSUE TITLE
[Wiki] [Docs] [Editor] Please provide documentation on "using the in-game Scenario Editor"

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#5700]
From my vigil as a Discord moderator of the server, I have noticed a lot of people making content using the in-game scenario editor. However, not many of the WML/Lua consultants have experience in using the scenario editor. Thus, can the scenario editor functionality be documented with the commonly used features?

Also, please label as Docs, Editor, Good First Issue, and Help.

**Comment by Pentarctagon** [comment]
The ingame help has already been updated to explain the scenario editor's features in the recent 1.14 and 1.15 releases.  Is this something else?

**Comment by knyghtmare** [comment]
> The ingame help has already been updated to explain the scenario editor's features in the recent 1.14 and 1.15 releases. Is this something else?

Yes, some more subsections under "Unit Tools" would be nice, particularly on the initialization of sides and assigning leader, recruit and/or recall lists, etc.

**Comment by stevecotton** [comment]
> Yes, some more subsections under "Unit Tools" would be nice, particularly on the initialization of sides and assigning leader, recruit and/or recall lists, etc.

Trouble is, the part of the new docs most relevant to those tasks is the line "<b>Warning: the Scenario Mode is buggy in 1.14.</b> Improving it is one of the main blockers for the 1.16 release, however in 1.14 its quality is far below that of the game and the terrain mode."

The recruit lists got ripped out of 1.16 in c8666a8a410ff0034a2c9bc5356ebfb80a15dd83.

**Comment by FrostBlade5** [comment]
As mentioned by stevecotton https://github.com/wesnoth/wesnoth/commit/c8666a8a410ff0034a2c9bc5356ebfb80a15dd83
The option to "asign recruit-list via the ingame scenario editor has been removed.

I cannot understand why you would remove such a usefull feature.
I often used this, basically in all of my maps, as i am not faimiliar with WML i am doing everything with the ingame editor.

Honestly i cant say how much of a downgrade it was to remove this functionallity.
Hopefully it will be re-added.

I never had any problems using the asign recruit list option, if it was unstable a warning to accept the risk would have been way better instead of completly eliminating this very useful feature.

Again, i do hope you will re-add this option as it was very useful.

**Comment by FrostBlade5** [comment]
https://postimg.cc/gallery/YLBYvMR

I created a little explenation on how to use the assign recruit list.
Maybe this might be of use for someone.
If you decide to add this useful feature back into the game.

**Comment by cooljeanius** [comment]
Ideally a single wiki page would also be helpful, too; searching the wiki now I'm finding a bunch of old GSoC pages about work on a scenario editor, but can't really tell which of them (if any) resulted in the creation of the Scenario Editor that actually exists

**Comment by stevecotton** [comment]
I've checked those pages for information, and marked them all as obsolete. It seems the scenario editor that actually exists started as Fendrin's work in 2013.

Before it had scenario editing support, the map editor had a basic feature to handle terrain maps embedded in .cfg files. So there's mention of .cfg files in src/editor long before any support for editing anything else about the scenario.

It would be nice to have a page on the Wiki similar to https://wiki.wesnoth.org/WesnothManual auto-generated from data/core/editor/help.cfg, and then editable pages for adding comments about it. Trouble with that is, that itself manual isn't auto-generated from data/core/help.cfg, rather it needs to be hand-crafted by editing doc/manual/manual.txt.

**Comment by stevecotton** [comment]
> https://postimg.cc/gallery/YLBYvMR
> 
> I created a little explenation on how to use the assign recruit list. Maybe this might be of use for someone. If you decide to add this useful feature back into the game.

Thanks @FrostBlade5, but your instructions require the user to manually edit the WML to set `faction=Custom`, and they would also need to manually edit the WML to fix any mistakes of accidentally adding an unwanted unit to the recruit list. While you may think you don't know WML, you clearly know more than you're giving yourself credit for.

**Comment by RyanZWhalen** [comment]
Hello! I'm a student from Codepath's A301 course. Would love to contribute to this issue if there's still elements that are left unresolved. Will take a look at everything that was done above, of course. Please let me know what else would be important for documentation.

**Comment by stevecotton** [comment]
We're currently in string-freeze for the 1.20 release, so can't add new documentation to the game at the moment, although it could be added to the Wiki. Our "Good first issue" issues assume someone who's familiar with playing the game is wanting to start developing it, or in the case of this one, someone who's familiar with playing scenarios, who wants to make their own via the editor's scenario support, is finding things that aren't documented - or that are documented, but don't work as documented. Fixing stuff to work as the documentation says it does can still be done during the string freeze.

**Comment by knyghtmare** [comment]
> We're currently in string-freeze for the 1.20 release, so can't add new documentation to the game at the moment, although it could be added to the Wiki. Our "Good first issue" issues assume someone who's familiar with playing the game is wanting to start developing it, or in the case of this one, someone who's familiar with playing scenarios, who wants to make their own via the editor's scenario support, is finding things that aren't documented - or that are documented, but don't work as documented. Fixing stuff to work as the documentation says it does can still be done during the string freeze.

@stevecotton the **key deliverable** is not to add these docs in the game, but rather **as a wiki page article**. It's not needed to be added to the game.

**Comment by knyghtmare** [comment]
Also, this issue has aged really well. We now have unit type creator feature inside the editor and that is missing a lot of documentation too. 
Things I would like to see in the documentation:
- How to create time area, and how to set up time area in the scenario
- How to define a side and then add it's customized recruit list
- how to make a spawned unit a leader type with own recruit list (oh, needs a verify check if the editor can support extra_recruit or not, probably no)
- how to spawn units on a hex in scenario editor
- important part: How to link scenario editor output to an actual scenario cfg file.  I have seen so many people struggle in this part.
- How to load an era's units to the editor lists and spawn those on scenario map in scenario editor map
